File: project/Core/Entities/Event.cs
using System.ComponentModel.DataAnnotations;

namespace Core.Entities
{
    public class Event : BaseEntity
    {
        public Event()
        {
            this.EnrolledUsers = new HashSet<UserEvent>();
            this.Favorites = new HashSet<Favorite>();
            this.Recents = new HashSet<Recent>();
            this.Links = new HashSet<Link>();
        }

        [Required]
        public string Title { get; set; }

        [Required]
        public string Description { get; set; }

        [Required]
        public string ImageUrl { get; set; }

        [Required]
        public DateTime Date { get; set; } = DateTime.UtcNow;

        [Required]
        public int Duration { get; set; }

        [Required]
        public string Location { get; set; }

        [Required]
        public string PresenterImage { get; set; }

        [Required]
        public string PresenterName { get; set; }

        [Required]
        public string PresenterRole { get; set; }

        [Required]
        public int CategoryId { get; set; }

        public Category Category { get; set; }

        [Required]
        public int LevelId { get; set; }

        public Level Level { get; set; }

        public ICollection<UserEvent> EnrolledUsers { get; set; }

        public ICollection<Favorite> Favorites { get; set; }

        public ICollection<Recent> Recents { get; set; }

        public ICollection<Link> Links { get; set; }
    }
}

File: project/client/src/app/shared/models/event.ts
import { ILink } from './link';

export interface IEvent {
  id: number;
  title: string;
  category: string;
  description: string;
  imageUrl: string;
  participants: number;
  date: string;
  formattedStartDate: string;
  formattedEndDate: string;
  duration: number;
  location: string;
  presenterImage: string;
  presenterName: string;
  presenterRole: string;
  links: ILink[];
}

export interface IBaseInputEvent {
  title: string;
  description: string;
  date: string;
  categoryId: number;
  levelId: number;
  image: any;
  presenterImage: any;
  duration: number;
  location: string;
  presenterName: string;
  presenterRole: string;
}

export interface ICreateEvent extends IBaseInputEvent {}

export interface IEditEvent extends IBaseInputEvent {
  id: number;
  imageUrl: string;
  presenterImageUrl: string;
}

export interface IEnrollUser {
  userId: string;
  eventId: number;
}

export interface ICreateFavorite {
  userId: string;
  eventId: number;
}

export interface ICreateRecent {
  userId: string;
  eventId: number;
}

export interface IEventShortDetails {
  id: number;
  title: string;
  date: string;
  formattedStartDate: string;
  formattedEndDate: string;
  location: string;
}
